luxjs/services/pagination.js

```
@@ -0,0 +1,96 @@
+            this.api = $lux.api(this.target);
+
+            if (recursive === true) this.recursive = true;
+        }
+
+        LuxPagination.prototype.getData = function(params, cb) {
+            // getData runs $lux.api.get() followed by the component's
+            // callback on the returned data or error.
+            // it's up to the component to handle the error.
+
+            this.params = params ? params : null;
+            if (cb) this.cb = cb;
+
+            this.api.get(null, params).then(function(data) {
+
+                if (data.error) return cb(data);
+
+                this.cb(data);
+                this.updateUrls(data);
+
+            }.bind(this));
+
+        };
+
+        LuxPagination.prototype.updateUrls = function(data) {
+            // updateUrls creates an object containing the most
+            // recent last and next links from the API
+
+            if (data && data.data && data.data.last) {
+                this.emitEvent();
+                this.urls = {
+                    last: data.data.last,
+                    next: data.data.next ? data.data.next : false
+                };
+                // If the recursive param was set to true this will
+                // request data using the 'next' link
+                if (this.recursive) this.loadMore();
+            }
+
+        };
+
+        LuxPagination.prototype.emitEvent = function() {
+            // emit event if more data available, the component can
+            // listen for it and choose how to deal with it
+            this.scope.$emit('moreData');
+        };
+
+        LuxPagination.prototype.loadMore = function() {
+            // loadMore applies new urls from updateUrls to the
+            // target object and makes another getData request.
+
+            if (this.urls.next === false) {
+                this.target.url = this.urls.last;
+            } else if (this.urls.next) {
+                this.target.url = this.urls.next;
+            }
```

What about if someone calls this when there is no more data by error? Perhaps we should throw an exception?

Also is it possible to pass a URL via the options parameter of `this.apt.get` rather than overwriting it in the target?